Simplify `use VERSION behaviour` on 5.37 onwards

leonerd opened the pull request:
(this description was all wrong but in any case I've now reverted it to a draft. We'll wait for 5.39 to apply it properly. I'll write something better by then)

Leont commented:
There are two «if (shortver >= SHORTVER(5, 35))»s in there, can't they be merged?

tonycoz commented on op.c:
```diff
@@ -7899,20 +7899,18 @@ Perl_utilize(pTHX_ int aver, I32 floor, OP *version, OP *idop, OP *arg)
+            PL_hints |= HINT_STRICT_REFS | HINT_EXPLICIT_STRICT_REFS |
+                HINT_STRICT_SUBS | HINT_EXPLICIT_STRICT_SUBS |
+                HINT_STRICT_VARS | HINT_EXPLICIT_STRICT_VARS;
+        } else if (shortver >= SHORTVER(5, 11)) {
```
The style guide in perlguts says to use uncuddled elses (also below)

leonerd replied:
Ohyes, I just copied that from Zefram's patch. Shall fix.

leonerd commented:
> There are two «if (shortver >= SHORTVER(5, 35))»s in there, can't they be merged?

They could, but that might make the logic harder to read. At the moment, there's a clear split between "fiddling with strict bits", followed by "apply warnings".

leonerd commented:
Actually this doesn't fix any back-compat. It just adds new behaviour. Probably best to save it for .39 after all.

jkeenan commented:
@leonerd, this p.r. now has merge conflicts.  Do you want to resolve those -- or would it be better to start fresh with a new p.r.?

leonerd committed:
Back-compat fixes for `use VERSION` side-effects on strictness

Original PR text by Zefram:

Until now, "use VERSION" has only set stricture bits that were not
already set explicitly.  That's unlike the treatment of feature flags
and warnings.  This changes it so that for new versions, v5.37 and above,
"use VERSION" will enable the stricture bits unconditionally, as if there
had been an explicit "use strict".  The meanings of "use v5.36" and below
remain unchanged: they still won't override explicitly-set stricture
status.  "use v5.37" and above counts as explicitly enabling stricture.

(from https://github.com/Perl/perl5/pull/20923)

leonerd committed:
WIP: fiddlings for 5.39 horizon instead

jkeenan commented:
@leonerd, this pull request has been in draft status for 12 months and has merge conflicts for 6 months.  If you intend to move forward with it, could you resolve conflicts, remove the draft status and re-push?  Thanks.
